.github/workflows/sast.yaml

```diff
@@ -0,0 +1,20 @@
+name: SAST
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    types: [opened, synchronize, reopened]
+jobs:
+  sonarcloud:
+    name: SonarCloud
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0  # Shallow clones should be disabled for a better relevancy of analysis
+      - name: SonarCloud Scan
+        uses: SonarSource/sonarcloud-github-action@master
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}  # Needed to get PR information, if any
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
```

Comment:
Q: when should this workflow execute? Are you aware, that this secret is never available with the pull_request trigger, due to protection mechanisms?

Reply:
This is 1:1 from the docs of SonarCloud, I trust that they validated this approach
https://github.com/SonarSource/sonarcloud-github-action?tab=readme-ov-file#usage

Reply:
Ok, interesting, [GitHub itself](https://securitylab.github.com/research/github-actions-preventing-pwn-requests/#:~:text=Due%20to%20the%20dangers%20inherent%20to%20automatic%20processing%20of%20PRs%2C%20GitHub%E2%80%99s%20standard%20pull_request%20workflow%20trigger%20by%20default%20prevents%20write%20permissions%20and%20secrets%20access%20to%20the%20target%20repository.) tells something else, but happy to have learned something new.